CMakeLists.txt

```diff
@@ -22,7 +22,7 @@ find_package(up-core-api REQUIRED)
 # This is the root CMakeLists.txt file; We can set project wide settings here
 # TODO: Is this needed?
 if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
-	set(CMAKE_CXX_STANDARD 17)
+	set(CMAKE_CXX_STANDARD 20)
```

Comment:
Should the changes to switch to C++20 move to a different PR? Or are we OK reviewing them here since the implementation of the callback utils requires it right now.

Reply:
The change to C++20 has been removed from this PR. New PRs to follow.